### PR TITLE
Tran 89 password validation feature

### PR DESCRIPTION
**Pull request by amysiv42**
This pull request introduces a new password validation module under `src/auth`, providing a configurable and robust password policy system. The changes include core implementation, configuration, error handling, and comprehensive tests for password validation logic. Additionally, workflow and script updates support running these new tests.

**Key changes:**

**Password Validation Module Implementation**
- Added a new password validation system with configurable policy, error codes, and validation functions in `src/auth`. This includes:
  - `password.ts` for policy configuration and environment variable support,
  - `constants/password.ts` for error codes and config limits,
  - `validators/password.ts` for validation logic,
  - `utils/password/error-message.ts` for human-readable error messages,
  - `types/password.ts` for strong typing.  
  [[1]](diffhunk://#diff-d4e57c061a89b0172aed70a64690e8f36092ab4a76319a727e14f05a6c2de029R1-R39) [[2]](diffhunk://#diff-d569a841e3a5e08d7c7a29e6bada476798ba5c9cb2fca7bfcce8fdb040f1e046R1-R21) [[3]](diffhunk://#diff-94c0437ea2dc0de739ad13b0e8ef1789edece47055b69a8580007546f22af592R1-R80) [[4]](diffhunk://#diff-b379861b32efebc1ad1949132b6ba6fcbfe59cfea8d64eae890a55c4cd33665cR1-R20) [[5]](diffhunk://#diff-9a6552ebb2bc5ca7a9b6d96fd71cf8cb9ed4fc63ae672570e9592179cdbe63c9R1-R26)

**Testing and Test Infrastructure**
- Added comprehensive unit tests for password validation logic, including edge cases and environment-based configuration, in `test/auth/password/length-limits.test.ts` and `test/auth/password/validator.test.ts`.  
  [[1]](diffhunk://#diff-9518409889ac75d60b91ea0f2e91484ea988559545193f81ac38a0cb0495bbbcR1-R138) [[2]](diffhunk://#diff-cbd70b2103c248deb28994e5d334bb3f600561ec09de4df2962580de758dcc56R1-R195)

**Project and Build Configuration**
- Introduced a dedicated `package.json` and `tsconfig.json` for the `auth` module to support TypeScript compilation and isolated builds.  
  [[1]](diffhunk://#diff-a7c482496dea0b28451afd9ea64861a00d22c9efc9be2b8a35953b5331986babR1-R19) [[2]](diffhunk://#diff-90bc6a3ff1238833ada8af115fc0887039a66c0f3c46bd845b9b587dda59570fR1-R18)

**Workflow and Script Updates**
- Updated the CI workflow to pass password policy environment variables to the test runner, and added a new `test:auth` script for targeted auth tests.  
  [[1]](diffhunk://#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR27-R31) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R15)

### CODE REVIEW

File: src/auth/src/utils/password-validator.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { PasswordPolicy, PasswordValidationResult } from '../types/password.js';
+
+
+export function validatePassword(password: string, policy: PasswordPolicy) : PasswordValidationResult {
```
Comment (sairutra) [inline src/auth/src/utils/password-validator.ts]: Would it be beneficial to split each case into different functions? to return amount of lines in case it grows + for testability + modularity

File: src/auth/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "build": "tsc -p tsconfig.json",
+    "clean": "rm -rf ../../dist/auth"
+  },
+  "devDependencies": {
+    "@types/node": "^20.19.25",
+    "tsx": "^4.21.0",
+    "typescript": "^5.9.3"
+  },
+  "dependencies": {
+    "fastify": "^4.29.1"
```
[the remark this answers is not in the document]
Comment (sairutra) [reply src/auth/package.json]: You are going to use fastify later?

File: test/auth/utils/password-validator.test.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import { validatePassword } from '../../../src/auth/src/utils/password-validator.js';
```
Comment (sairutra) [inline test/auth/utils/password-validator.test.ts]: Think about previous comment about modularity and testability here

File: src/auth/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (amysiv42) [reply src/auth/package.json]: Yes , I will. In this feature it is useless.  I will remove it.

File: src/auth/src/utils/password/error-message.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  const messages: Record<PasswordErrorCode, string> = {
+    [PasswordError.TOO_SHORT]: `Password must be at least ${policy.minLength} characters long.`,
+    [PasswordError.TOO_LONG]: `Password must be at most ${policy.maxLength} characters long.`,
+    [PasswordError.NO_UPPERCASE]: 'Password must have at least 1 uppercase letter.',
+    [PasswordError.NO_LOWERCASE]: 'Password must have at least 1 lowercase letter.',
+    [PasswordError.NO_NUMBER]: 'Password must have at least 1 number.',
+    [PasswordError.NO_SPECIAL]: 'Password must have at least 1 special character.',
+    [PasswordError.HAS_SPACE]: 'Password must not contain spaces.'
+  };
+  return(messages[errorCode]);
```
Comment (sairutra) [inline src/auth/src/utils/password/error-message.ts]: check if you can make it simpler, for example by directly mapping in constants.ts
